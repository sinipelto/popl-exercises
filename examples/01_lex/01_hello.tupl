message <- "Hello popl".
Print[message].
[ message, "again" ] | Print.
