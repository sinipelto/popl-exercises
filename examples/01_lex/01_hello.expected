LexToken(varIDENT,'message',1,0)
LexToken(LARROW,'<-',1,8)
LexToken(STRING_LITERAL,'Hello popl',1,11)
LexToken(DOT,'.',1,23)
LexToken(funcIDENT,'Print',2,25)
LexToken(LSQUARE,'[',2,30)
LexToken(varIDENT,'message',2,31)
LexToken(RSQUARE,']',2,38)
LexToken(DOT,'.',2,39)
LexToken(LSQUARE,'[',3,41)
LexToken(varIDENT,'message',3,43)
LexToken(COMMA,',',3,50)
LexToken(STRING_LITERAL,'again',3,52)
LexToken(RSQUARE,']',3,60)
LexToken(PIPE,'|',3,62)
LexToken(funcIDENT,'Print',3,64)
LexToken(DOT,'.',3,69)
